File: dungeons-1.21.11/src/main/java/fishmod/utils/config/FolderUtility.java
package fishmod.utils.config;

import fishmod.utils.Constants;
import fishmod.utils.debug.Debug;

import java.io.File;

public class FolderUtility {

    public static final String OLD_PATH = "./config/";
    public static final String CONFIG_PATH = "config/fishmod/";

    public static final String ADDONS_NAME = Constants.NAMESPACE + ".json";

    public static void init() {
        File file = new File(CONFIG_PATH);
        if (!file.exists()) {
            if (!file.mkdirs()) {
                Debug.LOGGER.error("Failed to create fishmod directory");
            }
        }
    }

}

File: dungeons-26.1.2/src/main/kotlin/fishmod/utils/config/FolderUtility.kt
package fishmod.utils.config

import fishmod.utils.Constants
import fishmod.utils.debug.Debug
import java.io.File

object FolderUtility {

    @JvmField
    val OLD_PATH: String = "./config/"

    @JvmField
    val CONFIG_PATH: String = "config/fishmod/"

    @JvmField
    val ADDONS_NAME: String = Constants.NAMESPACE + ".json"

    @JvmStatic
    fun init() {
        val file = File(CONFIG_PATH)
        if (!file.exists()) {
            if (!file.mkdirs()) {
                Debug.LOGGER.error("Failed to create fishmod directory")
            }
        }
    }

}
